lib/solr.js

```diff
@@ -570,21 +571,72 @@ Client.prototype.get = function(handler,query,callback){
    }
 
    var fullPath = [this.options.path,this.options.core,handler + '?' + parameters + '&wt=json']
-                              .filter(function(element){
-                                 return element;
-                              })
-                              .join('/');
+         .filter(function(element){
+            return element;
+         })
+         .join('/'),
+       approxUrlLength = 10 + this.options.host.length + (this.options.port+"").length + fullPath.length; // Buffer (10) accounts for protocol and special characters like ://, port colon, and initial slash etc
+
+   if (approxUrlLength <= this.options.get_max_request_entity_size) {
```

Comment:
Check if  this.options.get_max_request_entity_size is truthy first to allow users to bypass this feature if they need to. 
```if(this.options.get_max_request_entity_size && approxUrlLength <= this.options.get_max_request_entity_size) {```